src/content.tsx

```
@@ -23,3 +24,51 @@ if (window.location.href.startsWith('https://codeforces.com/problemset')) {
     }
   }
 }
+
+// for LeetCode 
+if (window.location.href.startsWith('https://zerotrac.github.io/leetcode_problem_rating')) {
+  try {
+    const initializeLeetCode = () => {
+      const resetButton = Array.from(document.querySelectorAll('button.el-button--danger')).find(
+        (btn) => btn.textContent?.trim().toLowerCase().includes('reset')
+      );
+      
+      if (!resetButton) {
+        setTimeout(initializeLeetCode, 100);
```

Recalling the function is a good way to avoid any issues that could happen during the first loading, but the number of recalling the same function should be limited (3 max), or an edge case could occur when the function keeps calling itself, resulting in an infinite loop.

Personally, I prefer to throw an error directly if the button is not loaded to avoid unnecessary complexities, but a limit by 2 or 3 is also fine for me.